modules/calib3d/src/usac/ransac_solvers.cpp

```diff
@@ -0,0 +1,629 @@
+                                termination_criteria.update(best_model, best_score.inlier_number);
+                            }
+                    }
+                } // end of if so far the best score
+            } // end loop of number of models
+        } // end main while loop
+
+        // if best model has 0 inliers then return fail
+        if (best_score.inlier_number == 0)
+            return false;
+
+        // polish final model
+        if (params.getFinalPolisher() != PolishingMethod::NonePolisher) {
+            Mat polished_model;
+            Score polisher_score;
+            if (model_polisher.polishSoFarTheBestModel (best_model, best_score,
+                                                    polished_model, polisher_score))
+                if (polisher_score.better(best_score)) {
+                    best_score = polisher_score;
+                    polished_model.copyTo(best_model);
+                }
+        }
+
+        // ================= here is ending ransac main implementation ===========================
+        std::vector<bool> inliers_mask;
+        if (params.getTrace() >= 1) {
+            inliers_mask = std::vector<bool>(points_size);
+            // get final inliers from the best model
+            quality.getInliers(best_model, inliers_mask);
+        }
+        // Store results
+        ransac_output = RansacOutput::create(best_model, inliers_mask,
+                static_cast<int>(std::chrono::duration_cast<std::chrono::microseconds>
+                (std::chrono::steady_clock::now() - begin_time).count()), best_score.score,
+                best_score.inlier_number, iters, -1, -1);
+        return true;
+    }
+};
+
+/*
+ * pts1, pts2 are matrices either N x a, N x b or a x N or b x N, where N > a and N > b
+ * output is matrix of size N x (a + b)
+ * return points_size = N
+ */
+int mergePoints (const Mat &pts1, const Mat &pts2, Mat &pts) {
+    if (pts1.rows > pts1.cols) {
+        // concatenate matrices Nx2 and Nx2 to Nx4
+        hconcat(pts1, pts2, pts);
+    } else {
+        hconcat(pts1.t(), pts2.t(), pts);
+    }
+    pts.convertTo(pts, CV_32F); // convert to float
+    return pts.rows;
+}
+
+Mat findHomography (InputArray srcPoints, InputArray dstPoints, int method, double thr,
+    OutputArray mask, const int maxIters, const double confidence) {
+
+    Mat points;
+    int points_size = mergePoints(srcPoints.getMat(), dstPoints.getMat(), points);
+
+    Ptr<Model> params = Model::create(thr, EstimationMethod ::Homography,
+                          SamplingMethod::Uniform, confidence, maxIters, ScoreMethod::MSAC);
+
+    params->setTrace(mask.needed());
+    params->setLocalOptimization(LocalOptimMethod ::InLORsc);
+    params->setPolisher(PolishingMethod ::LSQPolisher);
+    params->setVerifier(VerificationMethod ::SprtVerifier);
+
+    // https://docs.opencv.org/2.4/modules/core/doc/utility_and_system_functions_and_macros.html#getcputickcount
```

Comment:
This comment look unrelated to the code near it.

Reply:
fixed